src/main/java/io/vertx/ext/jdbc/impl/JDBCClientImpl.java

```diff
@@ -53,7 +53,7 @@
  */
 public class JDBCClientImpl implements JDBCClient, Closeable {
 
-  private static final Pattern HOST_AND_PORT_PATTERN = Pattern.compile("://(\\p{Alnum}+)(?::([0-9]+))?");
+  private static final Pattern HOST_AND_PORT_PATTERN = Pattern.compile("://([\\p{Alnum}.]+)(?::([0-9]+))?");
```

Comment:
How about reusing the pattern from sql client which has been test in many cases:

https://github.com/eclipse-vertx/vertx-sql-client/blob/4ba4aaef9f72b1adfd0dc4de1ead5f0455c60177/vertx-pg-client/src/main/java/io/vertx/pgclient/impl/PgConnectionUriParser.java#LL42C31-L42C31